src/webserver/DOSGuard.h

```diff
@@ -69,6 +73,333 @@ class BasicDOSGuard : public BaseDOSGuard
+            boost::asio::ip::address_v6_range subnetRangeHosts = subnet.hosts();
+
+            if (subnetError)
+            {
+                throw std::invalid_argument(subnetError.message());
+                return false;
+            }
+
+            boost::system::error_code error_code;
+            boost::asio::ip::address addr =
+                boost::asio::ip::make_address(ip, error_code);
+
+            if (error_code)
+            {
+                throw std::invalid_argument(error_code.message());
+                return false;
+            }
+
+            // Check if the address is in one of the hosts of subnet
+            if (subnetRangeHosts.find(addr.to_v6()) != subnetRangeHosts.end())
+            {
+                return true;
+            }
+        }
+
+        // Catch-All for incorrect cases
+        return false;
+    }
+
+    /**
+     * @brief Helper method that checks if an IP address is valid IPv4
+     *
+     * @param ip The IP address (explicit IPv4)
+     */
+    [[nodiscard]] bool
+    isValidIPv4(std::string const& ip) const
+    {
+        // CASE #1: Checks specific to detection of CIDR for IPv4
+        if (ip.find("/") != std::string::npos &&
+            checkParsedIPv4Address(ip.substr(0, ip.find("/"))) == true &&
+            checkCIDRIPv4(ip) == true)
+        {
+            return true;
+        }
+
+        // CASE #2: No CIDR, can check ip valid IP
+        else if (ip.find("/") == std::string::npos)
+        {
+            return checkParsedIPv4Address(ip);
+        }
+
+        // Catch-All for incorrect cases
+        return false;
+    }
+
+    /**
+     * @brief Helper method checks if an IP address is a valid IPv6 address
+     *
+     * @param ip The IP address (explicit IPv6), with/without a CIDR
+     */
+    [[nodiscard]] bool
+    isValidIPv6(std::string const& ip) const
+    {
+        // CASE #1: Checks specific to detection of CIDR for IPv6
+        if (ip.find("/") != std::string::npos &&
+            checkParsedIPv6Address(ip.substr(0, ip.find("/"))) == true &&
+            checkCIDRIPv6(ip) == true)
+        {
+            return true;
+        }
+
+        // CASE #2: General check to determine if valid IPv6 if no CIDR
+        else if (ip.find("/") == std::string::npos)
+        {
+            return checkParsedIPv6Address(ip);
+        }
+
+        // Catch-All for incorrect cases
+        return false;
+    }
+
+    /**
+     * @brief Wrapper method to check mixed bag of IP addresses and determine
+     * validity of IPv4 or IPv6
+     *
+     * @param ip The IP address to check, type unknown
+     */
+    [[nodiscard]] bool
+    checkValidityOfWhitelist(std::string const& ip) const
+    {
+        if (ip.find(".") != std::string::npos)
+        {
+            return isValidIPv4(ip);
+        }
+
+        else if (
+            ip.find(":") != std::string::npos ||
+            ip.find("::") != std::string::npos)
```

Comment:
If it can not find one ":", can it find two ":"?

Reply:
@cindyyan317 I had that to protect against the malformed input like ":::", which shouldn't exist and would pass iff I only left the former ":" condition